app/assets/javascripts/discourse/app/components/ignored-user-list.js

```diff
@@ -20,13 +22,13 @@ export default Component.extend({
       });
     },
     newIgnoredUser() {
-      const modal = showModal("ignore-duration-with-username", {
-        model: this.model,
-      });
-      modal.setProperties({
-        ignoredUsername: null,
-        onUserIgnored: (username) => {
-          this.items.addObject(username);
+      this.modal.show(IgnoreDurationModal, {
+        model: {
+          ...this.model,
```

Comment:
Is this kosher, or should I nest this under another named key? Feels a bit strange to mix with the keys below. 🤔

Reply:
Yeah, agreed - it's a little strange, and I suspect destructuring the user object will have some unexpected consequences.

IMO it would be better to do something like this:

```
model: {
  user: this.model,
  ignoredUsername: null,
  // ...
}
```
and then refactor the modal to look at `this.args.model.user.prop` instead of `this.args.model.prop`